mozci/console/commands/push.py

```diff
@@ -269,48 +276,96 @@ def handle(self):
-            elif classification == PushStatus.GOOD:
-                if push.backedout:
-                    good_backedout_pushes.append(push)
-                else:
-                    good_non_backedout_pushes.append(push)
-            else:
-                if push.backedout:
-                    unknown_backedout_pushes.append(push)
-                else:
-                    unknown_non_backedout_pushes.append(push)
+                    self.classifications[push] = PushStatus[
+                        artifact["push"]["classification"]
+                    ]
+                except TaskNotFound as e:
+                    self.line(
+                        f"<comment>Taskcluster task missing for {branch} {push.rev}</comment>"
+                    )
+                    self.errors[push] = e
+
+                except Exception as e:
+                    self.line(
+                        f"<error>Fetch failed on {branch} {push.rev}: {e}</error>"
+                    )
+                    self.errors[push] = e
+
+            # Advance the overall progress bar
+            progress.advance()
+
+        # Conclude the progress bar
+        progress.finish()
 
-        if classification_failed:
+        if self.errors:
             self.line(
-                f"<error>Failed to fetch or recalculate classification for {len(classification_failed)} out of {len(pushes)} pushes.</error>"
+                f"<error>Failed to fetch or recalculate classification for {len(self.errors)} out of {len(self.pushes)} pushes.</error>"
             )
-        self.line(
-            f"{len(bad_backedout_pushes)} out of {len(pushes)} pushes were backed-out by a sheriff and were classified as BAD."
-        )
-        self.line(
-            f"{len(bad_non_backedout_pushes)} out of {len(pushes)} pushes weren't backed-out by a sheriff and were classified as BAD."
-        )
-        self.line(
-            f"{len(good_backedout_pushes)} out of {len(pushes)} pushes were backed-out by a sheriff and were classified as GOOD."
-        )
-        self.line(
-            f"{len(good_non_backedout_pushes)} out of {len(pushes)} pushes weren't backed-out by a sheriff and were classified as GOOD."
-        )
-        self.line(
-            f"{len(unknown_backedout_pushes)} out of {len(pushes)} pushes were backed-out by a sheriff and were classified as UNKNOWN."
+
+        self.log_pushes(PushStatus.BAD, False)
+        self.log_pushes(PushStatus.BAD, True)
+        self.log_pushes(PushStatus.GOOD, False)
+        self.log_pushes(PushStatus.GOOD, True)
+        self.log_pushes(PushStatus.UNKNOWN, False)
+        self.log_pushes(PushStatus.UNKNOWN, True)
+
+        output = self.option("output")
+        if output:
+            # Build stats for CSV
+            with open(output, "w") as csvfile:
+                writer = csv.DictWriter(
+                    csvfile,
+                    fieldnames=[
+                        "push",
```

Comment:
We could remove this as the `revision` in a branch already uniquely identifies a push.

Reply:
Fixed in https://github.com/mozilla/mozci/pull/600/commits/e6d66b06884df15cc6bbd8a48b216b108c9af184